Kconfig.zephyr

```diff
@@ -344,6 +344,25 @@ config BUILD_OUTPUT_S19
 	  Build an S19 binary zephyr/zephyr.s19 in the build directory.
 	  The name of this file can be customized with CONFIG_KERNEL_BIN_NAME.
 
+config BUILD_OUTPUT_UF2
+	bool "Build a binary in UF2 format"
+	default n
+	depends on BUILD_OUTPUT_BIN
+	help
+	  Build a UF2 binary zephyr/zephyr.uf2 in the build directory.
+	  The name of this file can be customized with CONFIG_KERNEL_BIN_NAME.
+
+if BUILD_OUTPUT_UF2
+
+config BUILD_OUTPUT_UF2_FAMILY_ID
+	string "UF2 Device Family"
+	help
+	  UF2 bootloaders only accept UF2 files with a matching family ID.
+	  This can be either a hex, e.g. 0x68ed2b88, or well-known family
+	  name string.
```

Comment:
Based on this link: https://github.com/Microsoft/uf2#picking-numbers-at-random
I think it should be highlighted that it should not just be some hex numbers, but a good random number, and if a family name is used, then the known family name will be converted to the corresponding hex value.

Should we lock this field to a specific value if a SoC with a know family ID is used ?
Such as the nrf52, samd51, stm32, etc.

Reply:
I'm not sure about locking this field, but maybe *defaulting* the family ID value on those SoCs? If there's interest, I can update this to do that.

Reply:
@tejlmand Ok, I've just rebased and tweaked this a bit, and have added defaulting of the family ID config for the various SoC defconfig files, so for a given board, like the Seeeduino Xiao now in this PR, just enabling the UF2 build output is enough.